ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/GenesisState.java

```diff
@@ -133,17 +175,42 @@ private static void writeAccountsTo(
     target.persist(rootHeader);
   }
 
-  private static Hash calculateGenesisStateHash(final List<GenesisAccount> genesisAccounts) {
-    final ForestWorldStateKeyValueStorage stateStorage =
-        new ForestWorldStateKeyValueStorage(new InMemoryKeyValueStorage());
-    final WorldStatePreimageKeyValueStorage preimageStorage =
-        new WorldStatePreimageKeyValueStorage(new InMemoryKeyValueStorage());
-    final MutableWorldState worldState =
-        new ForestMutableWorldState(stateStorage, preimageStorage, EvmConfiguration.DEFAULT);
+  private static Hash calculateGenesisStateHash(
+      final DataStorageFormat dataStorageFormat, final List<GenesisAccount> genesisAccounts) {
+    final MutableWorldState worldState = loadWorldState(dataStorageFormat);
     writeAccountsTo(worldState, genesisAccounts, null);
     return worldState.rootHash();
   }
 
+  private static MutableWorldState loadWorldState(final DataStorageFormat dataStorageFormat) {
+    switch (dataStorageFormat) {
+      case BONSAI -> {
```

Comment:
We leak a lot of storage specific logic into GenesisState here.  Can we push this into DataStorageFormat itself, and have it take a KeyValueStorageProvider param?  I think that would encapsulate the storage-format-specific creation in the format itself.

Reply:
I updated my code